### PR TITLE
Coverity 1518612: Remove dead code

### PR DESCRIPTION
**Pull request by serrislew**
Fixes: #10332 

`out_end` cannot be nullptr, remove conditional 
- `*out_len` > 0 is always true

### CODE REVIEW

File: plugins/experimental/cookie_remap/strip.cc [hunk 1]
```diff
@@ -228,7 +228,7 @@ get_stripped(const char *in, ssize_t in_len, char *out, int *out_len, unsigned i
   retval = stripped_core(r, in_end, &w, out_end, flags);
 
   /* null terminate */
-  out_end += out_end ? 1 : 0;            /* undo decrement at start */
+  out_end++;                             /* undo decrement at start */
```
Comment (bneradt) [inline plugins/experimental/cookie_remap/strip.cc:231]: A simple enough change. But for my benefit, can you explain why out_end is always non-zero? It wasn't obvious to me reading the function that it would be. At 211 it's set to:

```cpp
out_end = out + *out_len;
```

I think char is signed with some compilers, so one or both of those could be negative, which complicates things.

Comment (serrislew) [reply plugins/experimental/cookie_remap/strip.cc:231]: maybe always was misleading but `get_stripped` only gets called here [`CookieJar::verify_value`] (https://github.pie.apple.com/ats/trafficserver/blob/master/plugins/experimental/cookie_remap/cookiejar.cc#L178) with the above code setting `out_len` > 0

Comment (bneradt) [reply plugins/experimental/cookie_remap/strip.cc:231]: OK, so this isn't an issue with your patch, but I'm concerned this code has a one byte buffer overrun potential:

1. The caller passes in a pointer to a buffer that will be filled called `out` and a length for the allocated space called `out_len`.
2. Then we set `out_end` accordingly: `out_end = out + *out_len;`
3. OK, so out_end points to the end of the allocated buffer.
4. Then we increment one past the end of that buffer: `out_end++;`
5. And then write into the `out` buffer pointed to by `w`.

The fact is, we are probably always well under `out_len`, so this doesn't cause problems. But isn’t that a one byte buffer overrun waiting to happen? 

In other words, the caller said "Don't write more than `out_len` bytes to `out`." Then we say, "OK, I'll write at most `out_len + 1` bytes to `out`.

Let me know whether I'm missing something.

Comment (bneradt) [reply plugins/experimental/cookie_remap/strip.cc:231]: @serrislew pointed out that earlier, in line 202, `out_len` was decremented, so this increment for `\0` should be safe per the incoming buffer.

**Review by bneradt — APPROVED** [review]

**Review by bneradt — COMMENTED** [review]
So is this OK if `out` is non-null, and `*out_len` is 0? In that case, we won't decement `*out_len` for a null terminator.

File: plugins/experimental/cookie_remap/strip.cc (hunk 1 above)
Comment (bneradt) [inline plugins/experimental/cookie_remap/strip.cc:231]: Should we add a parameter check at line 192 that if `out` is non-nullptr, then `*out_len > 0` ?

**Review by bneradt — CHANGES_REQUESTED** [review]

File: plugins/experimental/cookie_remap/strip.cc (hunk 1 above)
Comment (bneradt) [reply plugins/experimental/cookie_remap/strip.cc:231]: I think, in that case, there will be no space for the `out_end++`, right?

**Review by bneradt — APPROVED** [review]
